thor/main.py

```diff
@@ -43,25 +51,17 @@ def initialize_use_ray(config: Config) -> bool:
 
 @dataclass
 class CheckpointData:
-    stage: Literal[
-        "filter_observations",
-        "range_and_transform",
-        "cluster_and_link",
-        "initial_orbit_determination",
-        "differential_correction",
-        "recover_orbits",
-        "complete",
-    ]
-    filtered_observations: Optional[Observations] = None
-    transformed_detections: Optional[TransformedDetections] = None
-    clusters: Optional[Clusters] = None
-    cluster_members: Optional[ClusterMembers] = None
-    iod_orbits: Optional[FittedOrbits] = None
-    iod_orbit_members: Optional[FittedOrbitMembers] = None
-    od_orbits: Optional[FittedOrbits] = None
-    od_orbit_members: Optional[FittedOrbitMembers] = None
-    recovered_orbits: Optional[FittedOrbits] = None
-    recovered_orbit_members: Optional[FittedOrbitMembers] = None
+    stage: VALID_STAGES
+    filtered_observations: Observations = Observations.empty()
+    transformed_detections: TransformedDetections = TransformedDetections.empty()
+    clusters: Clusters = Clusters.empty()
+    cluster_members: ClusterMembers = ClusterMembers.empty()
+    iod_orbits: FittedOrbits = FittedOrbits.empty()
+    iod_orbit_members: FittedOrbitMembers = FittedOrbitMembers.empty()
+    od_orbits: FittedOrbits = FittedOrbits.empty()
+    od_orbit_members: FittedOrbitMembers = FittedOrbitMembers.empty()
+    recovered_orbits: FittedOrbits = FittedOrbits.empty()
+    recovered_orbit_members: FittedOrbitMembers = FittedOrbitMembers.empty()
```

Comment:
@akoumjian I had to remove the `Optional[Table] = None` here to make mypy happy since we never check if any of these are None in `link_test_orbit`. Instead, I've just instantiated them as empty tables. I don't like this as much but I think it's okay since it is the checkpoint stage name that determines what we have run and not if the tables are empty.

Reply:
Nevermind, this doesn't actually work.

Reply:
I can take a stab at the typing if you want.

Reply:
Sure, I'll see if I have any luck.